frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -310,6 +311,14 @@ export const applyVisualizationSettingsDataTransformations = (
       condition: settings["graph.y_axis.scale"] === "pow",
       fn: getKeyBasedDatasetTransform(seriesDataKeys, applySquareRootScaling),
     },
+    {
+      condition: isCategoryAxis(xAxisModel),
+      fn: getKeyBasedDatasetTransform([X_AXIS_DATA_KEY], value => {
+        return isCategoryAxis(xAxisModel) && value === null
```

Comment:
I don't think it's possible for a row value to be `undefined`, but just in case maybe we should use `==` instead?